src/geode/model/helpers/component_mesh_queries.cpp

```diff
@@ -137,4 +162,234 @@ namespace geode
+        }
+        return polygon_unique_vertices;
+    }
+
+    newPolyhedraAroundFacet block_mesh_polyhedra_from_surface_polygon(
+        const BRep& model,
+        const Block3D& block,
+        const Surface3D& surface,
+        index_t polygon_id )
+    {
+        OPENGEODE_EXCEPTION( model.is_boundary( surface, block )
+                                 || model.is_internal( surface, block ),
+            "[helpers::block_vertices_from_surface_polygon] The given surface "
+            "is neither boundary nor internal to the given block in the given "
+            "model." );
+        return block_mesh_polyhedra_from_unique_vertices_facet( model, block,
+            surface_polygon_unique_vertices( model, surface, polygon_id ) );
+    }
+
+    absl::InlinedVector< PolygonVertices, 2 >
+        block_vertices_from_surface_polygon( const BRep& model,
+            const Block3D& block,
+            const Surface3D& surface,
+            index_t polygon_id )
+    {
+        OPENGEODE_EXCEPTION( model.is_boundary( surface, block )
+                                 || model.is_internal( surface, block ),
+            "[helpers::block_vertices_from_surface_polygon] The given surface "
+            "is neither boundary nor internal to the given block in the given "
+            "model." );
+        const auto polygon_unique_vertices =
+            surface_polygon_unique_vertices( model, surface, polygon_id );
+        if( auto block_vertices =
+                block_unique_facet_vertices_from_surface_polygon(
+                    model, block, polygon_unique_vertices ) )
+        {
+            return { block_vertices.value() };
+        }
+
+        const auto facets_block_and_unique_vertices =
+            polygon_vertices_to_block_facets_vertices(
+                model, block, polygon_unique_vertices );
+        return polygon_to_block_vertices( polygon_unique_vertices,
+            facets_block_and_unique_vertices.first,
+            facets_block_and_unique_vertices.second );
+    }
+
+    BlockPolyhedraFacetVertices oriented_block_vertices_from_surface_polygon(
+        const BRep& model,
+        const Block3D& block,
+        const Surface3D& surface,
+        index_t polygon_id )
+    {
+        OPENGEODE_EXCEPTION( model.is_boundary( surface, block )
+                                 || model.is_internal( surface, block ),
+            "[helpers::block_vertices_from_surface_polygon] The given surface "
+            "is neither boundary nor internal to the given block in the given "
+            "model." );
+        const auto polygon_unique_vertices =
+            surface_polygon_unique_vertices( model, surface, polygon_id );
+        BlockPolyhedraFacetVertices p_to_block_vertices;
+        if( auto block_vertices =
+                block_unique_facet_vertices_from_surface_polygon(
+                    model, block, polygon_unique_vertices ) )
+        {
+            p_to_block_vertices.oriented_polyhedron_vertices =
```

Comment:
Why still oriented by default? Why not have the block_mesh_polyhedra_from_surface_polygon return a struct telling you which facet is oriented or not?